components/brave_today/browser/publishers_controller.cc

```diff
@@ -36,6 +38,45 @@
 
 namespace brave_news {
 
+namespace {
+mojom::Publisher* FindMatchPreferringLocale(
+    const Publishers& publishers,
+    const std::string& preferred_locale,
+    base::RepeatingCallback<bool(const mojom::Publisher&)> matcher) {
+  if (publishers.empty()) {
+    return nullptr;
+  }
+
+  mojom::Publisher* match = nullptr;
+  for (const auto& it : publishers) {
```

Comment:
Can we use destructuring here?

```suggestion
  for (const auto& [_, publisher] : publishers) {
```